### PR TITLE
Skip building SRPMS if already built

### PR DESCRIPTION
**Pull request by metalefty**
Fix for: https://github.com/AlmaLinux/build-system/issues/19

Replaces #57

### CODE REVIEW

**Review by Korulag — APPROVED** [review]

**Comment by metalefty** [comment]
@anfimovdm Could you also test this? I'll test this on my side, too.

File: build_node/builders/base_rpm_builder.py [hunk 1]
```diff
@@ -262,8 +263,13 @@ def build_packages(self, src_dir, spec_file=None):
...
                                           srpm_artifacts=True)
-        srpm_path = srpm_build_result.srpm
-        self.logger.info('src-RPM %s was successfully built', srpm_path)
+        if srpm_build_result:
+            srpm_path = srpm_build_result.srpm
+            self.logger.info('src-RPM %s was successfully built', srpm_path)
+        elif self.task.built_srpm_url:
+            srpm_path = built_srpm_url
+            self.logger.info('src-RPM %s is already built', built_srpm_url)
+
```
Comment (zklevsha) [inline build_node/builders/base_rpm_builder.py:272]: I guess we need to either
add **else** block with some kind of loging/exception or change **elif** to **else**

**Review by zklevsha — CHANGES_REQUESTED** [review]

File: build_node/builders/base_rpm_builder.py (hunk 1 above)
Comment (metalefty) [reply build_node/builders/base_rpm_builder.py:272]: Indeed, else block with some logging would be fine.

File: build_node/builders/base_rpm_builder.py [hunk 2]
```diff
@@ -248,10 +248,11 @@ def build_packages(self, src_dir, spec_file=None):
                                                      srpm_build=True)
         srpm_build_result = None
         try:
-            srpm_build_result = self.build_srpm(srpm_mock_config, src_dir,
-                                                srpm_result_dir,
-                                                spec_file=spec_file,
-                                                definitions=mock_defines)
+            if not self.built_srpm_url:
```
Comment (anfimovdm) [inline build_node/builders/base_rpm_builder.py:251]: it should be `self.task.built_srpm_url` instead

File: build_node/builders/base_rpm_builder.py [hunk 3]
```diff
@@ -262,8 +263,13 @@ def build_packages(self, src_dir, spec_file=None):
             if srpm_build_result:
                 self.save_build_artifacts(srpm_build_result,
                                           srpm_artifacts=True)
-        srpm_path = srpm_build_result.srpm
-        self.logger.info('src-RPM %s was successfully built', srpm_path)
+        if srpm_build_result:
+            srpm_path = srpm_build_result.srpm
+            self.logger.info('src-RPM %s was successfully built', srpm_path)
+        elif self.task.built_srpm_url:
+            srpm_path = built_srpm_url
```
Comment (anfimovdm) [inline build_node/builders/base_rpm_builder.py:270]: `srpm_path` is possibly unbound and `built_srpm_url` is undefined

File: build_node/builders/base_rpm_builder.py [hunk 4]
```diff
@@ -262,8 +263,13 @@ def build_packages(self, src_dir, spec_file=None):
...
+            self.logger.info('src-RPM %s is already built', built_srpm_url)
+
         excluded, reason = self.is_build_excluded(srpm_path)
```
Comment (anfimovdm) [inline build_node/builders/base_rpm_builder.py:273]: you should pass here path to SRPM instead of `built_srpm_url`

**Review by anfimovdm — CHANGES_REQUESTED** [review]

File: build_node/builders/base_rpm_builder.py (hunk 3 above)
Comment (metalefty) [reply build_node/builders/base_rpm_builder.py:270]: Right, I meant self.task.build_srpm here.

File: build_node/builders/base_rpm_builder.py (hunk 4 above)
Comment (metalefty) [reply build_node/builders/base_rpm_builder.py:273]: Indeed.

Comment (metalefty) [reply build_node/builders/base_rpm_builder.py:277]: Passed `os.path.join(src_dir, os.path.basename(self.task.built_srpm_url))`.  

However, built_srpm_url doesn't seem to be set from anywhere.

Comment (metalefty) [reply build_node/builders/base_rpm_builder.py:277]: built_srpm_url comes from web-server: https://chat.almalinux.org/almalinux/pl/ttm61gngd38zmrie9kmx984zao

**Review by zklevsha — APPROVED** [review]

**Comment by metalefty** [comment]
Closing due to unnecessity.

**Pull request closed without merging**